mirascope/base/calls.py

```diff
@@ -109,3 +111,40 @@ def _setup(
             tool_types = kwargs.pop("tools")
             kwargs["tools"] = [tool_type.tool_schema() for tool_type in tool_types]
         return kwargs, tool_types
+
+    @classmethod
+    def from_call(
+        cls, base_call_type: type[BaseCallT], call_params: BaseCallParams
+    ) -> type[BaseCallT]:
+        """Dynamically creates a new call class from a base model and a call.
+
+        Args:
+            cls: The call to use for the base model.
```

Comment:
you don't need to include `cls` here because it will automatically be injected by calling it from the class. Main thing is just the args the user needs to provide